parser-typechecker/src/Unison/Codebase/Branch.hs

```diff
@@ -341,8 +341,26 @@ deepEdits' b = go id b where
 
 merge :: forall m . Monad m => Branch m -> Branch m -> m (Branch m)
 merge (Branch x) (Branch y) =
-  Branch <$> Causal.threeWayMerge merge0 diff0 apply x y
+  Branch <$> Causal.threeWayMerge combine x y
  where
+  combine :: Maybe (Branch0 m) -> Branch0 m -> Branch0 m -> m (Branch0 m)
```

Comment:
The new `threeWayMerge` is parameterized on a function, `combine`, which takes the LCA if it exists, and the two descendants, and produces the merged result.